### PR TITLE
Update for next scippnexus

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Required changes for scipp/scippnexus#54. This will not pass tests until that is released.

There is a chicken-egg problem here, since scippnexus is not deprecating the old behavior, simply replacing it.

### CODE REVIEW

**Review by nvaytet — COMMENTED** [review]
Changes look fine to me, but I guess we need to require `scippnexus>=0.3` for the CI and the conda meta so this can work.

File: conda/meta.yaml [hunk 1]
```diff
@@ -37,15 +37,15 @@ requirements:
   host:
     - python
     - scipp {{ scipp }}
-    - scippnexus>=0.1
+    - scippnexus>=0.3
```
Comment (SimonHeybrock) [inline conda/meta.yaml:40]:
```suggestion
    - scippnexus>=0.3.1
```

File: conda/meta.yaml [hunk 2]
```diff
@@ -37,15 +37,15 @@ requirements:
...
     - tbb-devel=2020.2.*
   run:
     - h5py<3.2  # see https://github.com/h5py/h5py/issues/1880#issuecomment-823223154
     - numpy {{ numpy }}
     - pooch
     - python
     - scipp {{ scipp }}
-    - scippnexus>=0.1
+    - scippnexus>=0.3
```
Comment (SimonHeybrock) [inline conda/meta.yaml:48]:
```suggestion
    - scippnexus>=0.3.1
```

File: conda/meta.yaml [hunk 3]
```diff
@@ -37,15 +37,15 @@ requirements:
   host:
     - python
     - scipp {{ scipp }}
-    - scippnexus>=0.1
+    - scippnexus>=0.3.1
```
Comment (SimonHeybrock) [inline conda/meta.yaml:40]:
```suggestion
    - scippnexus>=0.3.2
```

File: conda/meta.yaml [hunk 4]
```diff
@@ -37,15 +37,15 @@ requirements:
...
     - tbb-devel=2020.2.*
   run:
     - h5py<3.2  # see https://github.com/h5py/h5py/issues/1880#issuecomment-823223154
     - numpy {{ numpy }}
     - pooch
     - python
     - scipp {{ scipp }}
-    - scippnexus>=0.1
+    - scippnexus>=0.3.1
```
Comment (SimonHeybrock) [inline conda/meta.yaml:48]:
```suggestion
    - scippnexus>=0.3.2
```

**Comment by SimonHeybrock** [comment]
For some reason CI keeps installing an old scippnexus, despite updated `meta.yml` (scippnexus>=0.3.2).

**Comment by SimonHeybrock** [comment]
Oh, it is probably the conda env caching getting in the way? Added a minimum version also to `.buildconfig/ci-.yml`.

**Pull request merged**